### DISCUSSION TITLE
Parallel API call support

### DISCUSSION THREAD

**Discussion by umarkhanTR** [pactumjs/pactum#303] [Q&A]
Hi
I'm pretty new to Pactum and JS world. How can we use Pactum for parallel API calls. Let's say I have an endpoint, DELETE which takes a query parameter of an entity to delete. There are 100 entities that I need to delete through this endpoint.

**Comment by ASaiAnudeep** [comment]
While Pactum itself doesn't directly support parallel API calls, there are workarounds you can use to achieve your goal of deleting 100 entities concurrently:

```js
const entities = [1, 2, 3, ..., 100]; // Your list of entity IDs

const promises = entities.map(entityId => {
  return pactum.spec()
    .delete(`http://your-api.com/entities/${entityId}`)
    .expectStatus(204); // Expect successful deletion
});

Promise.all(promises)
  .then(() => console.log("All entities deleted successfully!"))
  .catch(error => console.error("Error deleting entities:", error));

```
